internal/store/store.go

```diff
@@ -89,235 +106,268 @@ func customCATransport(ca []byte) http.RoundTripper {
 	}
 }
 
-func skipVerifyTransport() http.RoundTripper {
-	tlsClientConfig := &tls.Config{InsecureSkipVerify: true} //nolint:gosec // this is optional for some users
-	return &http.Transport{
-		ResponseHeaderTimeout: responseHeaderTimeout,
-		TLSHandshakeTimeout:   tlsHandshakeTimeout,
-		DialContext: (&net.Dialer{
-			Timeout:   timeout,
-			KeepAlive: keepAlive,
-		}).DialContext,
-		MaxIdleConns:          maxIdleConns,
-		MaxIdleConnsPerHost:   maxIdleConnsPerHost,
-		Proxy:                 http.ProxyFromEnvironment,
-		IdleConnTimeout:       idleConnTimeout,
-		ExpectContinueTimeout: expectContinueTimeout,
-		TLSClientConfig:       tlsClientConfig,
+func (s *Store) httpClient(httpTransport http.RoundTripper) (*http.Client, error) {
+	if s.username == "" || s.password == "" {
+		client := http.DefaultClient
+		client.Transport = httpTransport
+		return client, nil
 	}
-}
-
-type Config interface {
-	Service() string
-	PublicAPIKey() string
-	PrivateAPIKey() string
-	OpsManagerURL() string
-	OpsManagerCACertificate() string
-	OpsManagerSkipVerify() string
-	OpsManagerVersionManifestURL() string
-}
-
-func authenticatedClient(c Config) (*http.Client, error) {
 	t := &digest.Transport{
-		Username: c.PublicAPIKey(),
-		Password: c.PrivateAPIKey(),
-	}
-	if caCertificate := c.OpsManagerCACertificate(); caCertificate != "" {
-		dat, err := ioutil.ReadFile(caCertificate)
-		if err != nil {
-			return nil, err
-		}
-		t.Transport = customCATransport(dat)
-	} else if skipVerify := c.OpsManagerSkipVerify(); skipVerify == yes {
-		t.Transport = skipVerifyTransport()
-	} else {
-		t.Transport = http.DefaultTransport
+		Username: s.username,
+		Password: s.password,
 	}
-
+	t.Transport = httpTransport
 	return t.Client()
 }
 
-func defaultClient(c Config) (*http.Client, error) {
-	client := http.DefaultClient
-	if caCertificate := c.OpsManagerCACertificate(); caCertificate != "" {
-		dat, err := ioutil.ReadFile(caCertificate)
+func (s *Store) transport() (*http.Transport, error) {
+	switch {
+	case s.caCertificate != "":
```

Comment:
[q] In which scenario `s.caCertificate` would be not empty? are we using `caCertificate` somewhere?

Reply:
be default is not empty, you have to set the cli to use the custom ca